docs/config/server-options.md

```diff
@@ -139,10 +139,23 @@ Disable or configure HMR connection (in cases where the HMR websocket must use a
 
 Set `server.hmr.overlay` to `false` to disable the server error overlay.
 
-`clientPort` is an advanced option that overrides the port only on the client side, allowing you to serve the websocket on a different port than the client code looks for it on. Useful if you're using an SSL proxy in front of your dev server.
+`clientPort` is an advanced option that overrides the port only on the client side, allowing you to serve the websocket on a different port than the client code looks for it on.
 
 If specifying `server.hmr.server`, Vite will process HMR connection requests through the provided server. If not in middleware mode, Vite will attempt to process HMR connection requests through the existing server. This can be helpful when using self-signed certificates or when you want to expose Vite over a network on a single port.
 
+::: tip NOTE
+
+By default configuration, reverse proxies in front of Vite is expected to support proxying WebSocket. Especially in this case, when Vite HMR client fails to connect WebSocket, the client fallbacks to connect WebSocket directly to the Vite HMR server bypassing reverse proxies.
```

Comment:
I'm not sure I understand this note. I think it's saying that Vite has a fallback to support its WebSocket connection behind a proxy (i.e., when the proxy doesn't support WebSocket connections), but that fallback mechanism can cause error messages in the browser console. And the error message can be silenced with some Vite config (described below). Is that right?

Reply:
Yes, your understanding is correct.

Note that the fallback mechanism works if browser could access Vite's HMR Server.
For example, it does not work when Vite is running in docker and the port that Vite's HMR Server is listening is not forwarded.